### ISSUE TITLE
EPSG 3857 service returning 404s

### ISSUE THREAD

**Issue by sethjhs9** [nasa-gibs/onearth#183]
The EPSG 3857 service is returning incorrect tiles and 404 statuses. The bug is easy to see by scrolling/zooming in the NASA example: https://nasa-gibs.github.io/gibs-web-examples/examples/openlayers/webmercator-epsg3857.html

**Comment by jtroberts** [comment]
This was a GIBS server issue with the EPSG3857 reprojection endpoint that was resolved over the weekend. We'll assess whether we can make improvements in the OnEarth software to prevent the issue from reoccurring.

**Comment by sethjhs9** [comment]
Hey there,

Looks like this issue has returned. You can see the error easily here:
https://nasa-gibs.github.io/gibs-web-examples/examples/openlayers/webmercator-epsg3857.html

How can I help resolve this?

Cheers,
Seth

**Comment by lucianpls** [comment]
@jtroberts, is this due to the use of the ecache module and the disk filling up?

**Comment by jtroberts** [comment]
No, it's a server startup configuration issue when GIBS scales up new instances. The EPSG3857 endpoints use the reproject servers which rely on the EPSG4326 layer configurations to determine whether the layers are time-dynamic on startup. On rare occasions, the time configurations can't be read and the EPSG3857 endpoint configures that layer as static (not time-dynamic) and will serve tiles for the default time regardless of what is provided in the time parameter. The load balancer will distribute requests between good and bad servers (causing that slicing effect). A permanent fix is being developed and isn't in place yet.

**Comment by sethjhs9** [comment]
Looks like this behavior is happening again now for at least some of the SST layers. I don’t have any way link but this is the worldview view on the same layer: https://worldview.earthdata.nasa.gov/?v=-130.23220254985264,13.612503907831234,-112.85489894104501,42.920493576417265&l=Coastlines_15m,GHRSST_L4_MUR_Sea_Surface_Temperature&lg=true&t=2025-04-01-T22%3A18%3A35Z

**Comment by sethjhs9** [comment]
Nvm this is still visible here: https://nasa-gibs.github.io/gibs-web-examples/examples/openlayers/webmercator-epsg3857.html

How can I help out?

**Comment by sethjhs9** [comment]
This is happening again
I’d love to help out, is there any way I can?

Easily visible:  https://nasa-gibs.github.io/gibs-web-examples/examples/openlayers/webmercator-epsg3857.html

**Comment by sethjhs9** [comment]
Any way I can help out here?

I’m seeing this behavior now

**Comment by jtroberts** [comment]
There is an engineer looking into this but a long term solution is not yet available. The alerts have been helpful. I'm not sure if there is another way to further help without access to GIBS systems.

**Comment by sethjhs9** [comment]
Thanks @jtroberts, I’ll keep alerting you guys here 
Looks like there’s no more outage/degradation now. That last one lasted about a day

If something changes, let me know I’d love to help out and resolve this

**Comment by sethjhs9** [comment]
I see black tiles again, roughly 1 in 5 requests get a default black tile from the 3857 service

**Comment by sethjhs9** [comment]
One way to make this easier to mitigate without tricky infra changes would be to return 404s rather than black default tiles, I bet that would be easier to implement 
I'd love to help out if I can!

**Comment by jtroberts** [comment]
There was a healthcheck update to the GIBS EPSG3857 endpoints last week that should now catch and handle this problem if it occurs. Though it may take a few minutes for the system to refresh in those situations.

404s are returned when the tiles are out of range or are not available. However, in this case there are valid black tiles because GIBS is pulling tiles from the default time (ignoring the requested time) for the layer, which made it difficult for the original healthcheck to detect.

There is still work going on to better understand the root cause and prevent the issue from ever occurring.

**Comment by sethjhs9** [comment]
Thanks for the help here @jtroberts !
Keep me posted on how I can help out.
I just saw this problem occur again for a few minutes and then the service healed. Progress! If the default time tile was a 404 rather than a black tile with a 200 status, it would be super helpful but it sounds like that’s maybe harder to implement than fixing the underlying issue? I.e return 404 if the time variable is not provided

**Comment by jtroberts** [comment]
Glad to see that the system is automatically recovering. We have a another fix that we're currently testing and assessing whether it introduces any latency to tile requests.

Returning a 404 if no time variable is provided would break any non-time aware application (if I'm understanding your suggestion correctly). We wouldn't be able to support that. The default tile will use the latest available time if no time is provided in the request.
